packages/clerk-js/src/ui/signUp/SignUpVerifyEmailAddressWithMagicLink.tsx

```diff
@@ -57,15 +59,11 @@ export function SignUpVerifyEmailAddressWithMagicLink(): JSX.Element {
     } else if (ver.verifiedFromTheSameClient()) {
       showMagicLinkVerificationModal();
     } else {
-      await completeSignUpFlow(su);
-    }
-  };
-
-  const completeSignUpFlow = async (su: SignUpResource) => {
-    if (su.status === 'complete') {
-      return setActive({
-        session: su.createdSessionId,
-        beforeEmit: navigateAfterSignUp,
+      await completeSignUpFlow({
+        signUp: su,
+        signUpUrl: displayConfig.signUpUrl,
```

Comment:
@chanioxaris To alleviate the confusion around URLs for now, can you please remind me why do we need to pass signUpUrl in the `completeSignUpFlow` helper? 

Can we do `navigate('verify-email-address');` and `navigate('verify-phone-number')` in every flow?

Reply:
The `navigate('verify-email-address')` will only work if we are on the `sign-up` path. It won't work for example if we are at `sign-up/continue` -> `sign-up/continue/verify-email-address` as that's not the correct path

Reply:
I can see two options at this point:

Reply:
Discussed offline and decided to go with option B for the moment, as a routing refactoring will come up within the next weeks